src/wwwroot/app/services/VacancyService.js

```diff
@@ -16,13 +31,57 @@ export default class VacancyService {
       return _HttpService.get(additionalUrl);
    }
 
+   _saveNewTopicsToThesaurus(data) {
+      let promises = {};
+      each(data, (list, thesaurusName) => {
+         promises[thesaurusName] = _ThesaurusService.saveThesaurusTopics(thesaurusName, list);
+      });
+      return _$q.all(promises);
+   }
+
+   _onError(err) {
+      _LoggerService.error(err);
+      return _$q.reject(err);
+   }
+
    saveVacancy(entity) {
-      if (entity.id) {
-         const additionalUrl = VACANCY_URL + entity.id;
-         return _HttpService.put(additionalUrl, entity);
-      } else {
-         return _HttpService.post(VACANCY_URL, entity);
-      }
+      let newDates = [];
+      each(DATE_TYPE, (type) => {
+         newDates.push(utils.formatDateToServer(entity[type]));
+         each(newDates, (date) => {
```

Comment:
`each` inside `each`? really? And what it doing? Also you don't need `newDates` (as I understand)